build/syncset.go

```diff
@@ -388,27 +386,43 @@ func createSelectorSyncSet(resources []runtime.RawExtension) *hivev1.SelectorSyn
 				ResourceApplyMode: hivev1.SyncResourceApplyMode,
 				Resources:         resources,
 			},
-			ClusterDeploymentSelector: metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"api.openshift.com/managed": "true",
-				},
-			},
+			ClusterDeploymentSelector: selector,
 		},
 	}
 }
+
+type SyncSetResourcesByLabelSelector struct {
+	entries []SyncSetResource
+}
+
+type SyncSetResource struct {
+	labelSelector metav1.LabelSelector
+	objects       []runtime.RawExtension
+}
+
+func (s *SyncSetResourcesByLabelSelector) Add(labelSelector metav1.LabelSelector, object runtime.RawExtension) {
+	for i, entry := range s.entries {
+		if reflect.DeepEqual(entry.labelSelector, labelSelector) {
+			s.entries[i].objects = append(s.entries[i].objects, object)
+			return
+		}
+	}
+	s.entries = append(s.entries, SyncSetResource{labelSelector, []runtime.RawExtension{object}})
```

Comment:
Could you add some comments here to help explain the flow of what's happening, especially on line 406 and 410?

Reply:
moved into its own package and refactored the functions a bit so it should be clearer what they do.